executor/batch_point_get.go

```diff
@@ -139,6 +139,9 @@ func (e *BatchPointGetExec) Next(ctx context.Context, req *chunk.Chunk) error {
 		if err := e.initialize(ctx); err != nil {
 			return err
 		}
+		if e.lock {
+			e.updateDeltaForTableID(e.tblInfo.ID)
```

Comment:
Do we have data race risks on `TableDeltaMap` like https://github.com/pingcap/tidb/pull/19006 if there are multiple `PointGetExec` or `BatchPointGetExec`  with locks executing concurrently as data sources.

Reply:
Looks possible. I think we should add a lock for `TxnCtx.UpdateDeltaForTable`.